utils/logging/src/lib.rs

```diff
@@ -0,0 +1,362 @@
+// Pruned copy of crate rust log, without global logger
+// https://github.com/rust-lang-nursery/log #7a60286
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! Log traits live here, which are called throughout the library to provide useful information for
+//! debugging purposes.
+//!
+//! There is currently 2 ways to filter log messages. First one, by using compilation features, e.g "max_level_off".
+//! The second one, client-side by implementing check against Record Level field.
+//! Each module may have its own Logger or share one.
+
+use core::{cmp, fmt};
+
+static LOG_LEVEL_NAMES: [&str; 6] = ["GOSSIP", "TRACE", "DEBUG", "INFO", "WARN", "ERROR"];
+
+/// An enum representing the available verbosity levels of the logger.
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+pub enum Level {
+    /// Designates extremely verbose information, including gossip-induced messages
+    Gossip,
+    /// Designates very low priority, often extremely verbose, information
+    Trace,
+    /// Designates lower priority information
+    Debug,
+    /// Designates useful information
+    Info,
+    /// Designates hazardous situations
+    Warn,
+    /// Designates very serious errors
+    Error,
+}
+
+impl PartialOrd for Level {
```

Comment:
this seems to be equal to the derived PartailOrd for a generic enum. Is that the case? If yes why you coded it instead of just derive it?

Reply:
Agreed - I pulled a lot of this code from the LN rust library - they implemented this back in 2018 - maybe the default implementation was different back then. I derived PartialOrd and Ord and all the tests still pass.